olp-cpp-sdk-authentication/src/AuthenticationClientImpl.cpp

```diff
@@ -635,76 +631,48 @@ client::CancellationToken AuthenticationClientImpl::SignUpHereUser(
-  if (!settings_.network_request_handler) {
-    thread::ExecuteOrSchedule(settings_.task_scheduler, [callback] {
-      callback(
-          client::ApiError::NetworkConnection("Cannot sign out while offline"));
-    });
-    return client::CancellationToken();
-  }
-  std::weak_ptr<http::Network> weak_network(settings_.network_request_handler);
-  std::string url = settings_.token_endpoint_url;
-  url.append(kSignoutEndpoint);
-  http::NetworkRequest request(url);
-  auto network_settings =
-      http::NetworkSettings()
-          .WithTransferTimeout(settings_.retry_settings.timeout)
-          .WithConnectionTimeout(settings_.retry_settings.timeout)
-          .WithProxySettings(settings_.network_proxy_settings.get_value_or({}));
+  OLP_SDK_CORE_UNUSED(credentials);
+  using ResponseType = client::ApiResponse<SignOutResult, client::ApiError>;
 
-  request.WithVerb(http::NetworkRequest::HttpVerb::POST);
-  request.WithHeader(http::kAuthorizationHeader,
-                     GenerateBearerHeader(access_token));
-  request.WithHeader(http::kUserAgentHeader, http::kOlpSdkUserAgent);
-  request.WithSettings(std::move(network_settings));
+  auto sign_out_task =
+      [=](client::CancellationContext context) -> ResponseType {
+    if (!settings_.network_request_handler) {
+      return client::ApiError::NetworkConnection(
+          "Cannot sign out while offline");
+    }
 
-  std::shared_ptr<std::stringstream> payload =
-      std::make_shared<std::stringstream>();
-  auto send_outcome = settings_.network_request_handler->Send(
-      request, payload,
-      [callback, payload,
-       credentials](const http::NetworkResponse& network_response) {
-        auto response_status = network_response.GetStatus();
-        auto error_msg = network_response.GetError();
+    if (context.IsCancelled()) {
+      return client::ApiError::Cancelled();
+    }
 
-        if (response_status < 0) {
-          // Network error response not available
-          AuthenticationError error(response_status, error_msg);
-          callback(error);
-          return;
-        }
+    client::AuthenticationSettings auth_settings;
+    auth_settings.token_provider =
+        [&access_token](client::CancellationContext&) {
+          return client::OauthToken(access_token, kMaxTime);
+        };
 
-        auto document = std::make_shared<rapidjson::Document>();
-        rapidjson::IStreamWrapper stream(*payload);
-        document->ParseStream(stream);
+    client::OlpClient client = CreateOlpClient(settings_, auth_settings, false);
 
-        std::shared_ptr<SignOutResultImpl> resp_impl =
-            std::make_shared<SignOutResultImpl>(response_status, error_msg,
-                                                document);
-        SignOutResult response(resp_impl);
-        callback(response);
-      });
+    auto signout_response =
+        client.CallApi(kSignoutEndpoint, "POST", {}, {}, {}, {},
+                       kApplicationJson, std::move(context));
```

Comment:
kApplicationJson was not used before, and the appropriate header was not set. Are you sure this is ok as the REST API /logout does not accept any request body. I think in this case it is safer to not set any header, means leave parameter empty.

Reply:
done